### DISCUSSION TITLE
Store archive logs and artifacts via non dual stack endpoint into s3?

### DISCUSSION THREAD

**Discussion by Hiroki6** [argoproj/argo-workflows#13385] [Q&A] [answered]
I would like to store logs into s3 without dual stack endpoint.
As the context, we are setting network policy in our cluster and want to resolve s3 with a private ip with VPC endpoints.
However S3 VPC Endpoints [do not support dualstack](https://docs.aws.amazon.com/AmazonS3/latest/userguide/dual-stack-endpoints.html#dual-stack-endpoints-description). 
Therefore, I want to store logs with non-dual stack endpoint into s3.

As far as I see, the minio-go used by `github.com/argoproj/pkg/s3` enables the dual endpoint by default.
https://github.com/minio/minio-go/blob/d80dc8b8264d3391e255f1a1af529bdbe0d0e9cd/api.go#L83

However, I don't see any places to override this configuration.
https://github.com/argoproj/pkg/blob/42ec18a0638911de9d98b8faaabf2decfdf883cd/s3/s3.go
https://github.com/argoproj/argo-workflows/blob/b6d6403320157e614af2c14c2417433e55eadec1/workflow/artifacts/s3/s3.go#L27

**Comment by agilgur5** [comment, marked as answer]
Superseded by #13433 

Also please use permalinks when referring to specific files and especially line numbers. A link to the `main` branch can become outdated very quickly.
I've modified your Discussion and issue to use them
